### DISCUSSION TITLE
How to secure clientid using react

### DISCUSSION THREAD

**Discussion by billtlee** [Web3Auth/Web3Auth#404] [Help]
Hi,

I have a question about the clientid.  If we are using react to connect, wouldn't the clientid be accessible on the browser side through developer tools?  Even if we use environment variables, wouldn't it still be on the browser?  What is the best practice to hide it from the browser?

Thanks,

Bill

**Comment by himanshuchawla009** [comment]
You can keep clientId on the client side, it is totally safe.

**Reply by billtlee** [reply]
thx
